index.js

```diff
@@ -341,7 +342,40 @@ Feed.prototype.put = function (index, data, proof, cb) {
   this._putBuffer(index, this._codec.encode(data), proof, null, cb)
 }
 
-Feed.prototype.clear = function (start, end, opts, cb) {
+Feed.prototype.cancel = function (start, end) {  // TODO: use same argument scheme as download
+  if (!end) end = start + 1
+
+  // cancel these right away as .download does not wait for ready
+  for (var i = this._selections.length - 1; i >= 0; i--) {
+    var sel = this._selections[i]
+    if (start <= sel.start && sel.end <= end) {
+      this.undownload(sel)
+    }
+  }
+
+  // defer the last part until after ready as .get does that as well
+  if (this.opened) this._cancel(start, end)
+  else this._readyAndCancel(start, end)
+}
+
+Feed.prototype._cancel = function (start, end) {
```

Comment:
actually, we expect all the _get/_put etc methods to run after open so that is implied. let me just remove the comment